src/cfgparse.c

```diff
@@ -205,6 +205,12 @@ int str2listener(char *str, struct proxy *curproxy, struct bind_conf *bind_conf,
 			goto fail;
 		}
 	} /* end while(next) */
+#ifdef __linux__
+	if (global.tune.options & GTUNE_MPTCP) {
+		if (proto->sock_prot == IPPROTO_TCP)
+			proto->sock_prot = IPPROTO_MPTCP;
```

Comment:
mmh, did you even verify the modification you did?

It looks like here, you are only modifying the last `proto` (there is a loop above), after having created the listening socket. The modification should be done after (or in `str2sa_range()`) and before calling `create_listeners()`, no?

Please add a test.